force-app/main/default/classes/Email Recipes/InboundEmailHandlerRecipes.cls

```diff
@@ -0,0 +1,170 @@
+/**
+ * @description Demonstrates how to use the inboundEmailHandler
+ * interface to create custom logic and automation on the reception
+ * of an email. This class demonstrates saving the email
+ * to an EmailMessage Object along with Attachments.
+ * @group Email Recipes
+ * @see Safely, FilesRecipes
+ */
+public inherited sharing class InboundEmailHandlerRecipes implements Messaging.InboundEmailHandler {
+    public class InboundEmailHandlerRecipesException extends Exception {
+    }
+    /**
+     * @description    Messaging.InboundEmailHandler interface has one required
+     * method - handleInboundEmail. This method must return an
+     * Messaging.InboundEmailResult object, and you should take care to set that
+     * object's success property. This method is where you will write business
+     * logic to ... do whatever it is you want to do with the incoming email.
+     * Here you can attach the email to the contact record who sent it, a case
+     * or ... The sky's the limit.
+     * @param email    This is dependency injected by the system at runtime.
+     * @param envelope This is dependency injected by the system at runtime.
+     * @return         `Messaging.InboundEmailResult`
+     * @see FilesRecipes
+     */
+    public Messaging.InboundEmailResult handleInboundEmail(
+        Messaging.InboundEmail email,
+        Messaging.InboundEnvelope envelope
+    ) {
+        Messaging.InboundEmailResult result = new Messaging.InboundEmailresult();
+        String errorMessage = '';
+        try {
```

Comment:
General feedback on error handling: I see that you've defined a custom exception but your code is making little use of it (only for the contact name retrieval). Most of the errors you can trigger in this `try` block are silently ignored with return statements.
Although not critical, I think that it would make sense to stick to a single error handling behavior (throwing exceptions or silently ignoring errors).

Reply:
I see what you're saying. This was the result of a refactor to provide some additional test coverage to the Contact retrieval operation. I'm going to add some additional catch blocks here, because we don't have many examples of handling different types of exceptions in the codebase.

Reply:
addressed, but by doing an all exception capture.